src/app/models/jugaad_model.php

```diff
@@ -601,9 +601,9 @@ public function get_file_data($file_id, $template_meta, $user, $return_default =
             $field = $this->get_field_value($file_id, $name, $meta, $user);
             if ($field === false) {
                 if ($return_default) {
-                    $data[$name] = @$meta['default'] ?: $meta['name'];
+                    $data[$name] = @$meta['default'] ?: null;
```

Comment:
This will give null for non-optional fields too, if not set. Are you sure about that?